main_test.go

```diff
@@ -179,3 +184,228 @@ func TestParseConfigFromEnv(t *testing.T) {
 		t.Fatal("expected false but received true")
 	}
 }
+
+func TestParseConfigFromCredhub(t *testing.T) {
+	os.Clearenv()
+
+	credhubTs := testCredhubServer()
+	defer credhubTs.Close()
+	os.Setenv("CREDHUB_URL", credhubTs.URL)
+
+	uaaTs := uaaTestServer()
+	defer uaaTs.Close()
+	os.Setenv("UAA_ENDPOINT", uaaTs.URL)
+	os.Setenv("UAA_CLIENT_NAME", "client-name")
+	os.Setenv("UAA_CLIENT_SECRET", "client-secret")
+
+	config, err := parseConfig(logger)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if config.SecurityUserName != "securityUserName" {
```

Comment:
A table test for all of the simple string equality checks would compact this section nicely and make it a little more resilient to c/p errors.